### DISCUSSION TITLE
Issues with tiling manager (herbstluftwm)

### DISCUSSION THREAD

**Discussion by AckslD** [neovide/neovide#1486] [Q&A]
Hi! I started trying to use neovide which is simply amazing! I have a few small issues though and was wondering if someone might know what is going on. I'm not really sure if the issue is with neovide, herbstluftwm or the combination so this is why I open this as a discussion rather than an issue. I also think that some of the issues might be related, that's why I bundle them here.

# 1 Issue with initial layout
The first issue is that in some cases (not always) the size of the underlying neovim instance does not match the neovide window. If I make the window fullscreen and back again it is then the correct size. See the following screencast:

https://user-images.githubusercontent.com/23341710/184823248-18568ceb-4739-47fa-8153-29c02a9f563a.mp4

# 2 Setting the geometry when in floating mode
I sometimes want to use neovide in floating mode. So I have a rule that use floating instead of tiling if the window class is `floating`. This works but I'm not able to set the initial geometry. I tried the following:
* `neovide --x11-wm-class floating`: This gives a too large window which goes outside of my screen.
* `neovide --x11-wm-class floating --geometry 400x400`: This opens quickly but then crashes
* `neovide --x11-wm-class floating --maximized`: This opens a reasonable sized window but I'd like it slightly bigger.
* `neovide --x11-wm-class floating --maximized --geometry 400x400`: This also crashes.
See this:

https://user-images.githubusercontent.com/23341710/184825438-f6e655a0-04ec-4ef2-b18b-ad95b558de25.mp4

In the case when it crashes the error is:
```
thread 'main' panicked at 'Could not create surface', src/renderer/rendered_window.rs:75:6
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
and the log is [here](https://gist.github.com/AckslD/e94091465e35c5c0ef65d08ed53200ea).

# 3 Spawning neovide from keybinding
I have the following keybind in my herbstluftwm config:
```
hc keybind $Mod-Shift-Return spawn neovide
```
which works but the initial neovide instance is not interactable. For example if I try to press `i` then I don't get into insert mode. If I however make the window fullscreen and back again then it works normally again. See this:

https://user-images.githubusercontent.com/23341710/184825480-92212e76-a076-4a24-9eab-52b3028fd62a.mp4

# Software versions
neovide: 10.1
herbstluftwm: 0.9.5
os: arch linux 5.19.1-arch2-1

**Comment by MultisampledNight** [comment]
The crash at 2. seems interesting. Could you run Neovide with `--nofork --log` and edit your post to include both the log/a link to a gist which contains the log and the output (should be Rust panic-ish looking)?

**Reply by AckslD** [reply]
Hi @MultisampledNight! I updated the post to contain the error and log after the video in 2.

**Reply by AckslD** [reply]
@MultisampledNight let me know if you have some more thoughts on this and if I should open an issue instead?

**Reply by MultisampledNight** [reply]
I don't have any more thoughts except than the resize target size behaving potentially weird, feel free to open an issue. Thank you for your efforts though!
